pages/content.js

```diff
@@ -21,6 +21,25 @@ export default function Content({ currentCollections }) {
   const currentArticle = useStore((state) => state.currentArticle);
   const [popUp, setPopUp] = useState(false);
 
+  function highLight(event) {
+    if (!window.getSelection) return; //if no selection
+    if (window.getSelection().toString().length == 0) return;
+    if (event.detail === 2) return; // double mouse click
+    const range = window.getSelection().getRangeAt(0);
+    const span = document.createElement("span");
+    span.classList.add(".highlight");
+    span.style.backgroundColor = "#FFFF00";
+    span.appendChild(range.extractContents());
+    range.insertNode(span);
+
+    //remove highlight by double clicking on it
+    span.addEventListener("dblclick", (event) => {
+      span.style.userSelect = "none";
+      span.classList.remove(".highlight");
+      span.style.backgroundColor = "transparent";
+    });
```

Comment:
I see that you don't remove the injected span element. This is likely to cause bugs if you overlap highlighted sections. You should find a way to remove the span tags.

Reply:
Removed range and span! Fixed the last bug too :)